packages/block-library/src/navigation-link/edit.js

```diff
@@ -224,14 +225,25 @@ function Controls( { attributes, setAttributes, setIsLabelFieldFocused } ) {
 					label={ __( 'Link' ) }
 					value={ url ? safeDecodeURI( url ) : '' }
 					onChange={ ( urlValue ) => {
+						setAttributes( {
+							url: encodeURI( safeDecodeURI( urlValue ) ),
+						} );
+					} }
+					autoComplete="off"
+					type="url"
+					onFocus={ () => {
+						lastURLRef.current = url;
+						setIsEditingControl( true );
+					} }
+					onBlur={ () => {
+						// Defer the updateAttributes call to ensure entity connection isn't severed by accident.
 						updateAttributes(
-							{ url: urlValue },
+							{ url: ! url ? lastURLRef.current : url },
 							setAttributes,
-							attributes
+							{ ...attributes, url: lastURLRef.current }
 						);
+						setIsEditingControl( false );
```

Comment:
@getdave, what do you think about this?

The connection is severed after the first change on the trunk. This delays actual processing logic until the user is "done" editing, or if they have changed their mind.

Reply:
Good spot and I think we don't have much choice on this one. 

I'm surprised we don't already just use internal state (`useState`) to manage the value and then only "submit" it on blur.

Reply:
This also works, so decided to go without separate internal state.